File: packages/python-client-sdk/src/sdk/models/operations/mutation_products_update.py
import dataclasses
from dataclasses_json import dataclass_json
from enum import Enum
from sdk import utils
from typing import Optional


@dataclasses.dataclass
class MutationProductsUpdatePathParams:
    product_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'productId', 'style': 'simple', 'explode': False }})
    
class MutationProductsUpdateRequestBodyProductConfigSeatingConfigSeatingStrategyNameEnum(str, Enum):
    MONTHLY_ACTIVE_USER = "monthly_active_user"
    FIRST_COME_FIRST_SERVED = "first_come_first_served"


@dataclass_json
@dataclasses.dataclass
class MutationProductsUpdateRequestBodyProductConfigSeatingConfig:
    default_low_seat_warning_level_percent: float = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('default_low_seat_warning_level_percent') }})
    limited_overflow_seating_enabled: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('limited_overflow_seating_enabled') }})
    low_seat_warning_level_pct: float = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('low_seat_warning_level_pct') }})
    owner_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('owner_id') }})
    seating_strategy_name: MutationProductsUpdateRequestBodyProductConfigSeatingConfigSeatingStrategyNameEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('seating_strategy_name') }})
    default_seat_expiry_in_days: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('default_seat_expiry_in_days') }})
    seat_reservation_expiry_in_days: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('seat_reservation_expiry_in_days') }})
    

@dataclass_json
@dataclasses.dataclass
class MutationProductsUpdateRequestBodyProductConfig:
    contact_page_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_page_url') }})
    contact_sales_email: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_sales_email') }})
    contact_sales_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_sales_url') }})
    contact_support_email: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_support_email') }})
    contact_support_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_support_url') }})
    home_page_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('home_page_url') }})
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    is_setup_complete: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_setup_complete') }})
    on_access_denied_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_access_denied_url') }})
    on_access_granted_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_access_granted_url') }})
    on_no_seat_available_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_no_seat_available_url') }})
    on_no_subscriptions_found_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_no_subscriptions_found_url') }})
    on_subscription_canceled_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_subscription_canceled_url') }})
    on_subscription_not_found_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_subscription_not_found_url') }})
    on_subscription_not_ready_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_subscription_not_ready_url') }})
    on_subscription_suspended_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_subscription_suspended_url') }})
    owner_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('owner_id') }})
    privacy_notice_page_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('privacy_notice_page_url') }})
    product_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('product_name') }})
    publisher_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('publisher_name') }})
    seating_config: MutationProductsUpdateRequestBodyProductConfigSeatingConfig = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('seatingConfig') }})
    

@dataclass_json
@dataclasses.dataclass
class MutationProductsUpdateRequestBody:
    product_config: MutationProductsUpdateRequestBodyProductConfig = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('productConfig') }})
    

@dataclasses.dataclass
class MutationProductsUpdateRequest:
    path_params: MutationProductsUpdatePathParams = dataclasses.field()
    request: MutationProductsUpdateRequestBody = dataclasses.field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass_json
@dataclasses.dataclass
class MutationProductsUpdateDefaultApplicationJSONIssues:
    message: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    

@dataclass_json
@dataclasses.dataclass
class MutationProductsUpdateDefaultApplicationJSON:
    code: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('code') }})
    message: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    issues: Optional[list[MutationProductsUpdateDefaultApplicationJSONIssues]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('issues') }})
    
class MutationProductsUpdate200ApplicationJSONSeatingConfigSeatingStrategyNameEnum(str, Enum):
    MONTHLY_ACTIVE_USER = "monthly_active_user"
    FIRST_COME_FIRST_SERVED = "first_come_first_served"


@dataclass_json
@dataclasses.dataclass
class MutationProductsUpdate200ApplicationJSONSeatingConfig:
    default_low_seat_warning_level_percent: float = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('default_low_seat_warning_level_percent') }})
    limited_overflow_seating_enabled: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('limited_overflow_seating_enabled') }})
    low_seat_warning_level_pct: float = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('low_seat_warning_level_pct') }})
    owner_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('owner_id') }})
    seating_strategy_name: MutationProductsUpdate200ApplicationJSONSeatingConfigSeatingStrategyNameEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('seating_strategy_name') }})
    default_seat_expiry_in_days: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('default_seat_expiry_in_days') }})
    seat_reservation_expiry_in_days: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('seat_reservation_expiry_in_days') }})
    

@dataclass_json
@dataclasses.dataclass
class MutationProductsUpdate200ApplicationJSON:
    contact_page_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_page_url') }})
    contact_sales_email: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_sales_email') }})
    contact_sales_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_sales_url') }})
    contact_support_email: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_support_email') }})
    contact_support_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact_support_url') }})
    home_page_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('home_page_url') }})
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    is_setup_complete: bool = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_setup_complete') }})
    on_access_denied_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_access_denied_url') }})
    on_access_granted_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_access_granted_url') }})
    on_no_seat_available_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_no_seat_available_url') }})
    on_no_subscriptions_found_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_no_subscriptions_found_url') }})
    on_subscription_canceled_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_subscription_canceled_url') }})
    on_subscription_not_found_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_subscription_not_found_url') }})
    on_subscription_not_ready_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_subscription_not_ready_url') }})
    on_subscription_suspended_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('on_subscription_suspended_url') }})
    owner_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('owner_id') }})
    privacy_notice_page_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('privacy_notice_page_url') }})
    product_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('product_name') }})
    publisher_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('publisher_name') }})
    seating_config: MutationProductsUpdate200ApplicationJSONSeatingConfig = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('seatingConfig') }})
    

@dataclasses.dataclass
class MutationProductsUpdateResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    mutation_products_update_200_application_json_object: Optional[MutationProductsUpdate200ApplicationJSON] = dataclasses.field(default=None)
    mutation_products_update_default_application_json_object: Optional[MutationProductsUpdateDefaultApplicationJSON] = dataclasses.field(default=None)
    

File: packages/ts-client-sdk/src/sdk/models/operations/mutationproductsupdate.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class MutationProductsUpdatePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=productId" })
  productId: string;
}
export enum MutationProductsUpdateRequestBodyProductConfigSeatingConfigSeatingStrategyNameEnum {
    MonthlyActiveUser = "monthly_active_user",
    FirstComeFirstServed = "first_come_first_served"
}

export class MutationProductsUpdateRequestBodyProductConfigSeatingConfig extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=default_low_seat_warning_level_percent" })
  defaultLowSeatWarningLevelPercent: number;

  @SpeakeasyMetadata({ data: "json, name=default_seat_expiry_in_days" })
  defaultSeatExpiryInDays?: number;

  @SpeakeasyMetadata({ data: "json, name=limited_overflow_seating_enabled" })
  limitedOverflowSeatingEnabled: boolean;

  @SpeakeasyMetadata({ data: "json, name=low_seat_warning_level_pct" })
  lowSeatWarningLevelPct: number;

  @SpeakeasyMetadata({ data: "json, name=owner_id" })
  ownerId: string;

  @SpeakeasyMetadata({ data: "json, name=seat_reservation_expiry_in_days" })
  seatReservationExpiryInDays?: number;

  @SpeakeasyMetadata({ data: "json, name=seating_strategy_name" })
  seatingStrategyName: MutationProductsUpdateRequestBodyProductConfigSeatingConfigSeatingStrategyNameEnum;
}

export class MutationProductsUpdateRequestBodyProductConfig extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=contact_page_url" })
  contactPageUrl: string;

  @SpeakeasyMetadata({ data: "json, name=contact_sales_email" })
  contactSalesEmail: string;

  @SpeakeasyMetadata({ data: "json, name=contact_sales_url" })
  contactSalesUrl: string;

  @SpeakeasyMetadata({ data: "json, name=contact_support_email" })
  contactSupportEmail: string;

  @SpeakeasyMetadata({ data: "json, name=contact_support_url" })
  contactSupportUrl: string;

  @SpeakeasyMetadata({ data: "json, name=home_page_url" })
  homePageUrl: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=is_setup_complete" })
  isSetupComplete: boolean;

  @SpeakeasyMetadata({ data: "json, name=on_access_denied_url" })
  onAccessDeniedUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_access_granted_url" })
  onAccessGrantedUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_no_seat_available_url" })
  onNoSeatAvailableUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_no_subscriptions_found_url" })
  onNoSubscriptionsFoundUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_canceled_url" })
  onSubscriptionCanceledUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_not_found_url" })
  onSubscriptionNotFoundUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_not_ready_url" })
  onSubscriptionNotReadyUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_suspended_url" })
  onSubscriptionSuspendedUrl: string;

  @SpeakeasyMetadata({ data: "json, name=owner_id" })
  ownerId: string;

  @SpeakeasyMetadata({ data: "json, name=privacy_notice_page_url" })
  privacyNoticePageUrl: string;

  @SpeakeasyMetadata({ data: "json, name=product_name" })
  productName: string;

  @SpeakeasyMetadata({ data: "json, name=publisher_name" })
  publisherName: string;

  @SpeakeasyMetadata({ data: "json, name=seatingConfig" })
  seatingConfig: MutationProductsUpdateRequestBodyProductConfigSeatingConfig;
}

export class MutationProductsUpdateRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=productConfig" })
  productConfig: MutationProductsUpdateRequestBodyProductConfig;
}

export class MutationProductsUpdateRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: MutationProductsUpdatePathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: MutationProductsUpdateRequestBody;
}

export class MutationProductsUpdateDefaultApplicationJSONIssues extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export class MutationProductsUpdateDefaultApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=code" })
  code: string;

  @SpeakeasyMetadata({ data: "json, name=issues", elemType: MutationProductsUpdateDefaultApplicationJSONIssues })
  issues?: MutationProductsUpdateDefaultApplicationJSONIssues[];

  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}
export enum MutationProductsUpdate200ApplicationJSONSeatingConfigSeatingStrategyNameEnum {
    MonthlyActiveUser = "monthly_active_user",
    FirstComeFirstServed = "first_come_first_served"
}

export class MutationProductsUpdate200ApplicationJSONSeatingConfig extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=default_low_seat_warning_level_percent" })
  defaultLowSeatWarningLevelPercent: number;

  @SpeakeasyMetadata({ data: "json, name=default_seat_expiry_in_days" })
  defaultSeatExpiryInDays?: number;

  @SpeakeasyMetadata({ data: "json, name=limited_overflow_seating_enabled" })
  limitedOverflowSeatingEnabled: boolean;

  @SpeakeasyMetadata({ data: "json, name=low_seat_warning_level_pct" })
  lowSeatWarningLevelPct: number;

  @SpeakeasyMetadata({ data: "json, name=owner_id" })
  ownerId: string;

  @SpeakeasyMetadata({ data: "json, name=seat_reservation_expiry_in_days" })
  seatReservationExpiryInDays?: number;

  @SpeakeasyMetadata({ data: "json, name=seating_strategy_name" })
  seatingStrategyName: MutationProductsUpdate200ApplicationJSONSeatingConfigSeatingStrategyNameEnum;
}

export class MutationProductsUpdate200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=contact_page_url" })
  contactPageUrl: string;

  @SpeakeasyMetadata({ data: "json, name=contact_sales_email" })
  contactSalesEmail: string;

  @SpeakeasyMetadata({ data: "json, name=contact_sales_url" })
  contactSalesUrl: string;

  @SpeakeasyMetadata({ data: "json, name=contact_support_email" })
  contactSupportEmail: string;

  @SpeakeasyMetadata({ data: "json, name=contact_support_url" })
  contactSupportUrl: string;

  @SpeakeasyMetadata({ data: "json, name=home_page_url" })
  homePageUrl: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=is_setup_complete" })
  isSetupComplete: boolean;

  @SpeakeasyMetadata({ data: "json, name=on_access_denied_url" })
  onAccessDeniedUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_access_granted_url" })
  onAccessGrantedUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_no_seat_available_url" })
  onNoSeatAvailableUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_no_subscriptions_found_url" })
  onNoSubscriptionsFoundUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_canceled_url" })
  onSubscriptionCanceledUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_not_found_url" })
  onSubscriptionNotFoundUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_not_ready_url" })
  onSubscriptionNotReadyUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_suspended_url" })
  onSubscriptionSuspendedUrl: string;

  @SpeakeasyMetadata({ data: "json, name=owner_id" })
  ownerId: string;

  @SpeakeasyMetadata({ data: "json, name=privacy_notice_page_url" })
  privacyNoticePageUrl: string;

  @SpeakeasyMetadata({ data: "json, name=product_name" })
  productName: string;

  @SpeakeasyMetadata({ data: "json, name=publisher_name" })
  publisherName: string;

  @SpeakeasyMetadata({ data: "json, name=seatingConfig" })
  seatingConfig: MutationProductsUpdate200ApplicationJSONSeatingConfig;
}

export class MutationProductsUpdateResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  mutationProductsUpdate200ApplicationJSONObject?: MutationProductsUpdate200ApplicationJSON;

  @SpeakeasyMetadata()
  mutationProductsUpdateDefaultApplicationJSONObject?: MutationProductsUpdateDefaultApplicationJSON;
}